plugin/documents.py

```diff
@@ -519,16 +520,27 @@ def _on_hover_gutter_async(self, point: int) -> None:
                 on_navigate=lambda href: self._on_navigate(href, point))
 
     def on_text_command(self, command_name: str, args: Optional[dict]) -> Optional[Tuple[str, dict]]:
+        format_on_paste = self.view.settings().get('lsp_format_on_paste', userprefs().lsp_format_on_paste)
         if command_name == "auto_complete":
             self._auto_complete_triggered_manually = True
         elif command_name == "show_scope_name" and userprefs().semantic_highlighting:
             session = self.session_async("semanticTokensProvider")
             if session:
                 return ("lsp_show_scope_name", {})
+        elif command_name == 'paste_and_indent' and format_on_paste:
+            # if `lsp_format_on_paste` is enabled and paste_and_indent is run,
+            # it is easier to find the region to format when `paste` is invoked,
+            # so we intercept the `paste_and_indent` and replace it with the `paste` command.
+            if self.session_async("documentRangeFormattingProvider"):
```

Comment:
Minor but would prefer it like that so that all related logic is kept together and we also do less work.

```suggestion
        elif command_name == 'paste_and_indent'
            # if `lsp_format_on_paste` is enabled and paste_and_indent is run,
            # it is easier to find the region to format when `paste` is invoked,
            # so we intercept the `paste_and_indent` and replace it with the `paste` command.
            format_on_paste = self.view.settings().get('lsp_format_on_paste', userprefs().lsp_format_on_paste):
            if format_on_paste and self.session_async("documentRangeFormattingProvider"):
```

Same below.

Reply:
BTW. Most of this text is redundant since it's self-explanatory through the code itself. Basically only some variant of "it is easier to find the region to format when `paste` is invoked" is something that would be worth keeping.

Reply:
there is a syntax error, but I will push fix for it